src/CheckStrings.jl

```diff
@@ -0,0 +1,200 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+## Functions to check validity of UTF-8, UTF-16, and UTF-32 encoded strings,
+#  and also to return information necessary to convert to other encodings
+
+module CheckStrings
+
+using Encodings
+
+export check_string
```

Comment:
Let's just call it `check`; I'm a big fan of avoiding `_` where possible and letting the arguments play apart of indicating what a method does, i.e. `check(e::Encoding, dat)` == I'm checking the encoding/validity of `dat`.